common/predictive-text/worker/correction/distance-modeler.ts

```diff
@@ -586,7 +585,7 @@ namespace correction {
         // Build batches of same-cost entries.
         while(preprocessedQueue.count > 0) {
           let entry = preprocessedQueue.dequeue();
-          batch = batcher.checkAndAdd(entry);
+          let batch = batcher.checkAndAdd(entry);
```

Comment:
I'm moving `batch`'s declaration into deeper scopes, rather than relying on a single declaration, for one main reason:  the variable wasn't being reset for each loop iteration because it wasn't going out of scope.  This led to an interesting bug that took me a while to pin down.

While we _could_ just add two lines of code (well, one line in two spots) to reset the variable, it's probably better practice to declare copies of it in a manner that lets them naturally reinitialize due to going out of scope.

Reply:
Tighter scopes are usually better scopes 😁